video/out/wayland_common.c

```diff
@@ -962,25 +963,39 @@ static void handle_toplevel_config(void *data, struct xdg_toplevel *toplevel,
             wl->pending_vo_events |= VO_EVENT_LIVE_RESIZING;
             break;
         case XDG_TOPLEVEL_STATE_ACTIVATED:
+            /*
+             * We can partially detect when a window becomes unminimised.
+             * If we get an ACTIVATED state, we know it cannot be
+             * minimised, but we also see an ACTIVATED state when a
+             * window is focused, and the ACTIVATED is lost when a
+             * window loses focus, and when it is minimized, so we
+             * cannot use that to learn anything.
+             */
```

Comment:
Slight nitpick on this comment. A window losing focus doesn't necessarily mean it's no longer ACTIVATED (i.e. two mpv windows rendering side by side are both active even though only one is in focus).<sup>[1](https://github.com/wayland-project/wayland-protocols/blob/master/unstable/xdg-shell/xdg-shell-unstable-v6.xml#L712)</sup>

Reply:
I've cleaned up the comment.